File: Pascal's-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> res;
        if(numRows==1){
            res.push_back({1});
            return res;
        } 
        if(numRows==2){
            res.push_back({1});
            res.push_back({1,1});
            return res;
        }
        res.push_back({1});
        res.push_back({1,1});
        for(int i=2;i<numRows;i++){
            vector<int> temp;
            temp.push_back(1);
            for(int j=0;j<res.back().size()-1;j++){
                temp.push_back(res.back()[j]+res.back()[j+1]);
            }
            temp.push_back(1);
            res.push_back(temp);
        }
        return res;
    }
};

File: Day 1: Arrays/Java/Pascal's-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> res = new ArrayList<>();
        for(int i=0;i<numRows;i++){
            List<Integer> row = new ArrayList<>();
            if(i==0){
                row.add(1);
                res.add(row);
                continue;
            }
            List<Integer> lastrow = res.get(i-1);
            row.add(1);
            for(int j=0;j<lastrow.size()-1;j++){
                row.add(lastrow.get(j)+lastrow.get(j+1));
            }
            row.add(1);
            res.add(row);
        }
        return res;
    }
}


//O(n^2) time
//O(n) space
